### ISSUE TITLE
[improvement] Translation: automatic and manual language selection

### ISSUE THREAD

**Issue by AndreyMZ** [swagger-api/swagger-ui#2488]
**Problem description:**

https://github.com/swagger-api/swagger-ui/blob/1f2a5d7dd74131f0e9886b0f137ac20165aa90d2/dist/lang/translator.js#L6-L8
>     * To enable translation you should include one of language-files in your index.html
>     * after <script src='lang/translator.js' type='text/javascript'></script>.
>     * For example - <script src='lang/ru.js' type='text/javascript'></script>

This is unacceptable in today's internet. Every end user should see his language.
So:

1. `swagger-ui` should determine user's language automatically using the following variables (all of them in the specified order) and apply corresponding translation:
	
	1. [`navigator.languages`](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages)
	2. [`navigator.language`](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language)
	3. [`navigator.userLanguage`](https://msdn.microsoft.com/library/ms534713.aspx)
	
2. End user should be able to change the language using `swagger-ui`.

**Additional information:**
swagger-ui version affected: commit [1f2a5d7](https://github.com/swagger-api/swagger-ui/commit/1f2a5d7dd74131f0e9886b0f137ac20165aa90d2)
swagger file: not required

**Comment by heldersepu** [comment]
Not a fan this idea.
Let me start by saying that english is not my first language, not even my second, and I'm fine with everything (code/programing related) been in english ...


This is not an eCommerce platform where we try to reach a wide audience of end users, this is a tool for programmers, Anyone using swagger is familiar with basic english like the reserved words of JavaScript.

**Comment by AndreyMZ** [comment]
Swagger UI supported translation (in version 2.x): [swagger-ui/dist/lang/translator.js](https://github.com/swagger-api/swagger-ui/blob/c7439c79137e1e42dc67998f97d710996ca42ce6/dist/lang/translator.js).
And now the lack of translation support is listed as a known issue: [swagger-ui/README.md#known-issues](https://github.com/swagger-api/swagger-ui/blob/859edbf7f7139544751ad19c82c86013a7d6cd48/README.md#known-issues)

@heldersepu, as far as I understand, you want to say that Swagger UI should not support translation.
I am not going to argue with this, let the maintainers decide. I just want to note that you will not get any worse if translation support will be implemented properly (see item 3 of my suggestion above), you will be able to switch back to English.

When I said "End user should be able to change the language using swagger-ui", I meant he/she should be able to change it once for all. (Technically, the selected language should be remembered, e.g. using a cookie.)

**Comment by webron** [comment]
Yes, we do need to support translations, not sure the mentioned approach is the right one. However, we don't have the capacity to implement this right now.

**Comment by sivan-r** [comment]
Hi,
Is there any update regarding this issue?

**Comment by AlexMayants** [comment]
I am integrating Swagger UI (`swagger-ui-react`) into our company's application and ran into the lack of internationalization support. I think, the library should at least give developers an option to provide translations for static strings which are displayed in the interface. In this case language detection may be left at the discretion of the host application.

**Comment by cn-src** [comment]
How is the problem progressing?
You can first provide a way to extend the language pack, users can implement it themselves.

**Comment by fenomeno83** [comment]
I've created a my custom solution:
https://github.com/fenomeno83/SwaggerGlobalization

**Comment by kenspirit** [comment]
My solutions is different.  My focus is on providing multilingual schema instead of simply translating those labels/buttons which I considered less important.  Detail please check the issue #7917

**Comment by vizardkill** [comment]
up

**Comment by wardi** [comment]
Looking for some direction here: what would be the correct approach to provide translations for the text that is generated by swagger-ui?

I'm using some unpleasant CSS hacks to work around the issue now, but would much prefer doing this a cleaner way. Our team might be able to help submit a PR for such this change.
